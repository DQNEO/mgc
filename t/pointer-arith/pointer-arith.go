package main

import (
	"fmt"
	"unsafe"
)

func f0() {
	var c byte = '3'
	var d byte = '4'
	cp := &c
	*cp = '1'
	fmt.Printf("%c\n", c)
	fmt.Printf("%c\n", d)
}

func f1() {
	buf := []byte("123456789")
	addr := &buf[0]
	fmt.Printf("%s\n", buf)
	p := uintptr(unsafe.Pointer(addr))
	p = p + 2
	up := unsafe.Pointer(p)
	var pb *byte
	pb = (*byte)(up)
	*pb = '1'
	fmt.Printf("%s\n", buf)
}

func main() {
	f0()
	f1()
}
