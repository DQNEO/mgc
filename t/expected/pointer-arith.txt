1
4
123456789
121456789
